drivers/flash/Kconfig.stm32

```diff
@@ -14,11 +14,12 @@ menuconfig SOC_FLASH_STM32
 	select FLASH_HAS_DRIVER_ENABLED
 	default y
 	select FLASH_PAGE_LAYOUT if SOC_SERIES_STM32F0X
+	select FLASH_PAGE_LAYOUT if SOC_SERIES_STM32F3X
 	select FLASH_PAGE_LAYOUT if SOC_SERIES_STM32F4X
 	select FLASH_PAGE_LAYOUT if SOC_SERIES_STM32F7X
 	select FLASH_PAGE_LAYOUT if SOC_SERIES_STM32L4X
 	select FLASH_HAS_PAGE_LAYOUT if SOC_SERIES_STM32F0X
-	select FLASH_HAS_PAGE_LAYOUT if SOC_SERIES_STM32F4X
```

Comment:
This change seems spurious

Reply:
Indeed, was not intended. Txs for catch up! Fixed